src/test/architecture/test_architecture.py

```diff
@@ -312,3 +313,579 @@ def test_verify_callback_uniqueness(self, mocker, caplog):
+  - /callback_group_0
+- executor_type: single_threaded_executor
+  executor_name: executor_1
+  callback_group_names:
+  - /callback_group_1
+nodes:
+- node_name: /node_0
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_1
+    callback_names:
+    - /callback_0
+    - /callback_1
+  callbacks:
+  - callback_name: /callback_0
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_1
+    callback_type: subscription_callback
+    topic_name: /changed_topic
+    symbol: sub
+  publishes:
+  - topic_name: /topic_0
+    callback_names:
+    - /callback_0
+  subscribes:
+  - topic_name: /changed_topic
+    callback_name: /callback_1
+- node_name: /node_1
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_0
+    callback_names:
+    - /callback_2
+    - /callback_3
+  callbacks:
+  - callback_name: /callback_2
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_3
+    callback_type: subscription_callback
+    topic_name: /topic_0
+    symbol: sub
+  publishes:
+  - topic_name: /changed_topic
+    callback_names:
+    - /callback_2
+  subscribes:
+  - topic_name: /topic_0
+    callback_name: /callback_3"""
+
+        mocker.patch('builtins.open', mocker.mock_open(read_data=self.renaming_architecture_text))
+        arch = Architecture('yaml', 'architecture.yaml')
+
+        arch.rename_topic('/topic_1', '/changed_topic')
+
+        mocker.patch('builtins.open', mocker.mock_open(read_data=renamed_architecture_text))
+        arch_expected = Architecture('yaml', 'architecture.yaml')
+
+        assert set(arch_expected.topic_names) == set(arch.topic_names)
+
+        # check this
+        # publishes:
+        # - topic_name: /topic_0
+        #   callback_names:
+        #   - /callback_0
+        # subscribes:
+        # - topic_name: /changed_topic
+        #   callback_name: /callback_1
+        assert arch_expected.get_node('/node_0').get_publisher('/topic_0').topic_name \
+            == arch.get_node('/node_0').get_publisher('/topic_0').topic_name
+        assert arch_expected.get_node('/node_1').get_publisher('/changed_topic').topic_name \
+            == arch.get_node('/node_1').get_publisher('/changed_topic').topic_name
+
+        # check this
+        # publishes:
+        # - topic_name: /changed_topic
+        #   callback_names:
+        #   - /callback_2
+        # subscribes:
+        # - topic_name: /topic_0
+        #   callback_name: /callback_3"""
+        assert arch_expected.get_node('/node_0').get_subscription('/changed_topic').topic_name \
+            == arch.get_node('/node_0').get_subscription('/changed_topic').topic_name
+        assert arch_expected.get_node('/node_1').get_subscription('/topic_0').topic_name \
+            == arch.get_node('/node_1').get_subscription('/topic_0').topic_name
```

Comment:
Because the return value of anitecture object is a ValueObject, it can be evaluated as follows.
```suggestion
        assert arch.nodes == arch_expected.nodes
        assert arch.communications == arch_expected.communications
        assert arch.paths == arch_expected.paths
        assert arch.executors == arch_expected.executors
```

Reply:
Fixed it in [370ca2d](https://github.com/tier4/CARET_analyze/pull/156/commits/370ca2da6a861c0b72cd6f1cfdded510c16ec622)